src/Contracts/HasFakeData.php

```diff
@@ -0,0 +1,15 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Saloon\Contracts;
+
+use Saloon\Http\PendingRequest;
+
+interface HasFakeData
+{
+    /**
+     * Get the fake data for the mocked request.
+     */
+    public function getFakeData(PendingRequest $request): mixed;
```

Comment:
maybe replace return type by `array|string|MockResponse|Fixture` - or could this be anything else?

Reply:
No that's it I think. It's just a bit long in the implementation:

```php
/**
 * Get the fake data for the mocked request.
 */
public function getFakeData(PendingRequest $request): array|string|MockResponse|Fixture
{
    return ['status' => 'pending'];
}
```

Reply:
I prefer long explicit return types in interfaces. They can always be narrowed down in the implementation, so the following would be "PHPStan-safe", I guess:

```php
public function getFakeData(PendingRequest $request): array
{
    return ['status' => 'pending'];
}
```

should be possible since PHP 7.4, see [Covariant Returns and Contravariant Parameters](https://wiki.php.net/rfc/covariant-returns-and-contravariant-parameters)

Reply:
Fixed in e6dd5f9